### PR TITLE
:boom:feat(lld): Nfts collection refactor

### PR DESCRIPTION
**Pull request by LucasWerey**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 
- [x] **Impact of the changes:** 
  - No entryPoint set for the refactor so no visible change
  - All integration tests inside LLD 
  - Jest config

### 📝 Description

Refactor of the NFTs collection table in order to prepare the integration of other chains. 
Also there is a known issue with uuid so I've added a custom resolver for jest test as it's done here : https://github.com/microsoft/accessibility-insights-web/pull/5421/commits/9ad4e618019298d82732d49d00aafb846fb6bac7
Jest config has been fixed too to work with our project. The way we defined the router in testUtils will change when upgrading to last react-routes package version. eg: `<Switch>` will become `<Routes>`

We can now use all provided hooks from https://github.com/testing-library/jest-dom?tab=readme-ov-file#with-another-jest-compatible-expect

The collection on the top is the new one and the bottom one is the current one 

https://github.com/LedgerHQ/ledger-live/assets/73439207/51ed97ac-1dc5-4c91-a2c3-0e95c4df96cd

Screenshot 2024-07-02 at 16 40 31

### ❓ Context

- **JIRA or GitHub link**: [LIVE-12460](https://ledgerhq.atlassian.net/browse/LIVE-12460)

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-12460]: https://ledgerhq.atlassian.net/browse/LIVE-12460?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-desktop/src/newArch/Collectibles/components/Collection/TableRow/index.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      <>
+        <Skeleton width={42} minHeight={24} barHeight={10} show={props.isLoading}>
+          {isNFTRow(props) && (
+            <Text ff="Inter|SemiBold" color="palette.text.shade100" fontSize={4} textAlign="right">
+              {props.numberOfNfts || 0}
+            </Text>
+          )}
+          {isOrdinalsRow(props) && null}
+        </Skeleton>
+      </>
```
Comment (KVNLS) [inline apps/ledger-live-desktop/src/newArch/Collectibles/components/Collection/TableRow/index.tsx:45]: Same here, do we need the Fragment?

File: apps/ledger-live-desktop/src/newArch/Collectibles/__integration__/shared.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  <Switch>
+    <Route exact path="/" render={() => <NftCollection account={account} />} />
+    <Route path="/account/:id/nft-collection" render={() => <NFTGallery />} />
+  </Switch>
+);
+
+const NftCollectionTestBase = () => (
+  <>
+    <NftCollectionNavigation />
+  </>
```
Comment (KVNLS) [inline apps/ledger-live-desktop/src/newArch/Collectibles/__integration__/shared.tsx:22]: Do we need the Fragment?

File: apps/ledger-live-desktop/src/newArch/Collectibles/components/Collection/EmptyCollection.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                  ff="Inter|SemiBold"
+                  onClick={() => {
+                    openURL(supportLinkByTokenType.nfts);
+                    track(`More info on Manage nfts tokens`);
+                  }}
+                  label={t("tokensList.link")}
+                />
+              </Text>
+            </Placeholder>
+          </>
```
[the remark this answers is not in the document]
Comment (LucasWerey) [reply apps/ledger-live-desktop/src/newArch/Collectibles/components/Collection/EmptyCollection.tsx:61]: Removed ✅

File: apps/ledger-live-desktop/src/newArch/Collectibles/components/Collection/TableRow/index.tsx (hunk 1 above)
Comment (LucasWerey) [reply apps/ledger-live-desktop/src/newArch/Collectibles/components/Collection/TableRow/index.tsx:45]: Removed ✅

File: apps/ledger-live-desktop/src/newArch/Collectibles/__integration__/shared.tsx (hunk 2 above)
Comment (LucasWerey) [reply apps/ledger-live-desktop/src/newArch/Collectibles/__integration__/shared.tsx:22]: Removed ✅

File: apps/ledger-live-desktop/src/newArch/Collectibles/components/Collection/TableRow/TokenTitle.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import { Text } from "@ledgerhq/react-ui";
+
+type Props = {
+  tokenName: string | string[];
+  isLoading: boolean;
+};
+
+const TokenTitle: React.FC<Props> = ({ tokenName, isLoading }) => {
+  return (
+    <>
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-desktop/src/newArch/Collectibles/components/Collection/TableRow/TokenTitle.tsx:12]: you can remove `<> </>`

File: apps/ledger-live-desktop/src/newArch/Collectibles/Nfts/Collection/index.tsx [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+const NftItem: React.FC<NftItemProps> = ({
+  contract,
+  tokenId,
+  currencyId,
+  account,
+  numberOfNfts,
+  onClick,
+}) => {
+  const { metadata, status } = useNftMetadata(contract, tokenId, currencyId);
+
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-desktop/src/newArch/Collectibles/Nfts/Collection/index.tsx:57]: as you use several times `status === FieldStatus.Loading` you can use a const

File: apps/ledger-live-desktop/src/newArch/Collectibles/types/Media.ts [hunk 6]
```diff
@@ -16,18 +16,20 @@ type BaseMediaProps = BaseProps & {
   previewUri?: string | undefined;
   originalUri?: string | undefined;
   mediaType: string | undefined;
-  setUseFallback: (a: boolean) => void;
+  setUseFallback?: (a: boolean) => void;
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-desktop/src/newArch/Collectibles/types/Media.ts:19]: not a good practice to use single char const :)

File: apps/ledger-live-desktop/src/newArch/Collectibles/Nfts/Collection/useNftCollectionModel.tsx [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+            ),
+            action: onReceive,
+          },
+          {
+            element: <HeaderActions textKey="NFT.collections.galleryCTA" />,
+            action: onOpenGallery,
+          },
+        ]
+      : [];
+  }, [nftsInTheCollection.length, onReceive, onOpenGallery]);
```
[the remark this answers is not in the document]
Comment (LucasWerey) [reply apps/ledger-live-desktop/src/newArch/Collectibles/Nfts/Collection/useNftCollectionModel.tsx:99]: Moved inside the view

File: apps/ledger-live-desktop/src/newArch/Collectibles/utils/collectionUtils.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import { NFT, ProtoNFT } from "@ledgerhq/types-live";
+
+export const filterHiddenCollections = (
+  collections: (ProtoNFT | NFT)[] | Record<string, (ProtoNFT | NFT)[]>,
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-desktop/src/newArch/Collectibles/utils/collectionUtils.ts:4]: maybe you can create a sub type for `ProtoNFT | NFT`

File: apps/ledger-live-desktop/src/newArch/Collectibles/components/Collection/TableRow/TokenTitle.tsx (hunk 4 above)
Comment (LucasWerey) [reply apps/ledger-live-desktop/src/newArch/Collectibles/components/Collection/TableRow/TokenTitle.tsx:12]: Done ✅

File: apps/ledger-live-desktop/src/newArch/Collectibles/Nfts/Collection/index.tsx (hunk 5 above)
Comment (LucasWerey) [reply apps/ledger-live-desktop/src/newArch/Collectibles/Nfts/Collection/index.tsx:58]: Done ✅

File: apps/ledger-live-desktop/src/newArch/Collectibles/utils/collectionUtils.ts (hunk 8 above)
Comment (LucasWerey) [reply apps/ledger-live-desktop/src/newArch/Collectibles/utils/collectionUtils.ts:4]: Done ✅

File: apps/ledger-live-desktop/src/newArch/Collectibles/types/Media.ts (hunk 6 above)
Comment (LucasWerey) [reply apps/ledger-live-desktop/src/newArch/Collectibles/types/Media.ts:19]: Changed for `(useFallBack: boolean)`

File: apps/ledger-live-desktop/.unimportedrc.json [hunk 9]
```diff
@@ -19,10 +19,18 @@
     "node_modules/**",
     "src/generate-cryptoassets-md.test.ts",
     "src/generate-cryptoassets-md.ts",
+
     "src/newArch/Collectibles/**",
-    "src/newArch/WalletSync/**"
+    "src/newArch/WalletSync/**",
+    "src/newArch/components/ContextMenu/**"
```
Comment (LucasWerey) [inline apps/ledger-live-desktop/.unimportedrc.json:25]: Needed since contextMenu is used inside Collectibles folder which is not used for the moment

**Review by mcayuelas-ledger — APPROVED** [review]
LGTM

**Review by KVNLS — APPROVED** [review]

**Pull request merged**